Documentation/project-docs/changing-corelib.md

```diff
@@ -0,0 +1,22 @@
+Changing or adding new public APIs to System.Private.CoreLib
+============================================================
+
+Many of the CoreFX libraries type-forward their public APIs to the implementations in System.Private.CoreLib.
+- The CoreFX build uses System.Private.CoreLib via Microsoft.TargetingPack.Private.CoreCLR Nuget package.
+- Some of the CoreFX libraries are not built in the CoreFX repository. For example, System.Runtime.Loader.dll is purely a facade and type-forwards everything to System.Private.CoreLib. These libraries are built and published through a separate process.
+- Hence, when adding a new public API to System.Private.CoreLib or changing the behavior of the existing public APIs in System.Private.CoreLib, changes must be staged to ensure that new prerequisites are published before they are used.
+
+**Staging the changes**
+
+Make the changes in both CoreCLR and CoreFX
+- System.Private.CoreLib implementation changes should should be made in CoreCLR repo
```

Comment:
Small typo - "should should".

Reply:
Thanks, @danmosemsft has PR with further improvements for this doc: https://github.com/dotnet/coreclr/pull/21220 . I have asked him to fix this one as well.